### PR TITLE
Avoid validation of Jupyter Server providers

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #13894

### CODE REVIEW

File: src/kernels/jupyter/connection/serverUriStorage.ts [hunk 1]
```diff
@@ -372,16 +370,7 @@ class NewStorage {
                     displayName: item.displayName || uri,
                     provider: item.serverHandle
                 };
-                if (!validate) {
-                    entries.push(server);
-                    return;
-                }
-                try {
-                    await this.jupyterPickerRegistration.getJupyterServerUri(item.serverHandle, true);
```
Comment (DonJayamanne) [inline src/kernels/jupyter/connection/serverUriStorage.ts:380]: This is only required in one calling place, removed this and move the validation to the calling code.
Thereby removing a cycling reference and more layers.
This also makes listing of Jupyter Server much faster, as we no longer validate each server, which in turn waits for all related extensions to completely activate.

File: src/kernels/jupyter/connection/serverUriStorage.ts [hunk 2]
```diff
@@ -350,17 +334,17 @@ class NewStorage {
             })
             .catch(noop));
     }
-    public async getAll(validate = true): Promise<IJupyterServerUriEntry[]> {
-        return this.getAllImpl(validate).then((items) => items.sort((a, b) => b.time - a.time));
+    public async getAll(): Promise<IJupyterServerUriEntry[]> {
+        return this.getAllImpl().then((items) => items.sort((a, b) => b.time - a.time));
     }
     public async clear(): Promise<void> {
-        const all = await this.getAllImpl(false);
```
Comment (DonJayamanne) [inline src/kernels/jupyter/connection/serverUriStorage.ts:357]: `false` is the only value supported now.

File: src/kernels/jupyter/connection/serverUriStorage.unit.test.ts [hunk 3]
```diff
@@ -219,35 +218,30 @@ suite('Server Uri Storage', async () => {
                 .sort((a, b) => a.time - b.time)
                 .map((a) => {
                     return {
-                        displayName: a.displayName,
```
Comment (DonJayamanne) [inline src/kernels/jupyter/connection/serverUriStorage.unit.test.ts:222]: `displayName` is deprecated

File: src/kernels/jupyter/types.ts [hunk 4]
```diff
@@ -184,6 +184,7 @@ export interface IJupyterServerUriEntry {
     time: number;
     /**
      * An optional display name to show for this server as opposed to just the Uri
+     * @deprecated Used only for migration of display names into the User Provided Server list. Else other providers will have the Display Names.
```
Comment (DonJayamanne) [inline src/kernels/jupyter/types.ts:187]: `displayName` is deprecated, only used for migration of data from one storage into another

**Review by lramos15 — APPROVED** [review]

**Pull request merged**